Dalamud/Interface/Internal/InterfaceManager.cs

```diff
@@ -235,25 +234,42 @@ public IntPtr GameWindowHandle
     /// <summary>
     /// Dispose of managed and unmanaged resources.
     /// </summary>
-    public void Dispose()
+    void IInternalDisposableService.DisposeService()
     {
+        // Unload hooks from the framework thread if possible.
+        // The functions being unhooked are mostly called from the main thread, so unhooking here would avoid any
+        // chance of unhooking a function that currently is being called.
         if (Service<Framework>.GetNullable() is { } framework)
```

Comment:
Could this ever not be the case? If so, does that mean something is off with unload order? It seems to me that we would always want to avoid calling ClearHooks() off the framework thread, but please correct me if I'm missing something.

Reply:
`RunOnFrameworkThread` will do nothing if it's called after `Framework.Destroy`, so it doesn't matter. Otherwise, when initiated from /xldev "Unload Dalamud", framework should still be running, so that function actually does things. Anyway, added more comments on that.